connector/oidc/oidc.go

```diff
@@ -265,6 +275,12 @@ func (c *oidcConnector) LoginURL(s connector.Scopes, callbackURL, state string)
 	if s.OfflineAccess {
 		opts = append(opts, oauth2.AccessTypeOffline, oauth2.SetAuthURLParam("prompt", c.promptType))
 	}
+
+	if c.pkceVerifier != "" {
+		c.pkceVerifier = oauth2.GenerateVerifier()
+		opts = append(opts, oauth2.S256ChallengeOption(c.pkceVerifier))
```

Comment:
S256 can be unsupported by the provider. It is required to either clarify that Dex only supports the S256 challenge or provide support for others. 

We can rename the option `EnablePKCE` to `PKCEChallenge` instead of the boolean variable and let users specify the type of challenge they want to use.